### PR TITLE
[AETHER-1265] Integrate Java profiler in TOST env

### PR DESCRIPTION
**Pull request by pierventre**
Fix an issue in the atomix-agent that does prevent passing a different gc algorithm. Adds a new dockerfile that allows to build profiler enabled images.

### CODE REVIEW

File: dist/src/main/resources/bin/atomix-agent [hunk 1]
```diff
@@ -13,8 +13,14 @@ fi
...
 
+# Pass gc and other useful jvm opts through the env variable
+JAVA_OPTS="${JAVA_OPTS:--XX:+UseG1GC -XX:MaxGCPauseMillis=200}"
+if [ ! -z "$ATOMIX_YOURKIT" ]; then
+  JAVA_OPTS+=" -agentpath:$ATOMIX_ROOT/libyjpagent.so=listen=all"
+fi
+export JAVA_OPTS=${JAVA_OPTS}
+
 java \
-  -XX:+UseConcMarkSweepGC \
```
Comment (charlesmcchan) [inline dist/src/main/resources/bin/atomix-agent:17]: Is this removed intentionally? It seems irrelevant to the profiler change.

Comment (pierventre) [reply dist/src/main/resources/bin/atomix-agent:17]: Done intentionally - otherwise we cannot pass through the JAVA_OPTS a different gc

**Review by kuujo — APPROVED** [review]
